src/nsidc/iceflow/itrf/converter.py

```diff
@@ -69,17 +71,50 @@ def transform_itrf(
             if not plate:
                 plate = plate_name(Point(chunk.longitude.mean(), chunk.latitude.mean()))
             plate_model_step = (
-                f"+step +inv +init={target_itrf}:{plate} +t_epoch={target_epoch} "
+                # Perform coordinate propagation to the given epoch using the
+                # provided plate motion model (PMM).  An example is given in the
+                # message of this commit:
+                # https://github.com/OSGeo/PROJ/commit/403f930355926aced5caba5bfbcc230ad152cf86
+                f"+step +init={target_itrf}:{plate} +t_epoch={target_epoch} "
             )
 
+        itrf_transformation_step = ""
+        if source_itrf != target_itrf:
+            # This performs a helmert transform (see
+            # https://proj.org/en/9.4/operations/transformations/helmert.html). `+init=ITRF2014:ITRF2008`
+            # looks up the ITRF2008 helmert transformation step in the ITRF2014
+            # data file (see e.g.,
+            # https://github.com/OSGeo/PROJ/blob/master/data/ITRF2014). The
+            # `+inv` reverses the transformation. So `+init=ITRF2014:ITRF2008`
+            # performs a helmert transform from ITRF2008 to ITRF2014.
+            itrf_transformation_step = f"+step +inv +init={target_itrf}:{source_itrf} "
+
         pipeline = (
+            # This initializes the pipeline and declares the use of the WGS84
+            # ellipsoid for all of the following steps. See
+            # https://proj.org/en/9.5/operations/pipeline.html.
             f"+proj=pipeline +ellps=WGS84 "
+            # Performs unit conversion from lon/lat degrees to radians.
+            # TODO: This step appears to be unnecessary. Removing it does not appear to
+            # affect the output. The following steps require that the
+            # coordinates be geodedic, which could be radians or degrees.
             f"+step +proj=unitconvert +xy_in=deg +xy_out=rad "
+            # This step explicitly sets the projection as lat/lon. It won't
+            # change the coordinates, but they will be identified as geodetic,
+            # which is necessary for the next steps.
             f"+step +proj=latlon "
```

Comment:
This doesn't do anything aside from declare the coordinates to be geodetic, which is required by the subsequent steps. Not technically necessary but maybe better to be explicit in this case?